truss/templates/server/common/truss_server.py

```diff
@@ -131,7 +132,17 @@ def predict(
             body = json.loads(body_raw)
 
         # calls ModelWrapper.__call__, which runs validate, preprocess, predict, and postprocess
-        response: Dict = asyncio.run(model(body, headers=dict(request.headers.items())))
+        response: Union[Dict, Generator] = asyncio.run(
+            model(
+                body,
+                headers={key.lower(): value for key, value in request.headers.items()},
+            )
+        )
+
+        # In the case that the model returns a Generator object, return a
+        # StreamingResponse instead.
+        if isinstance(response, Generator):
+            return StreamingResponse(response, media_type="application/octet-stream")
```

Comment:
can you add a comment on how media type differs from content-type? That's the first thing that popped into my mind and i'm not sure if it conflates with the msgpack implementation